src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -141,6 +142,18 @@ exports.pushTopLevelContextObject = function(fiber : Fiber, context : Object, di
 function processChildContext(fiber : Fiber, parentContext : Object, isReconciling : boolean): Object {
   const instance = fiber.stateNode;
   const childContextTypes = fiber.type.childContextTypes;
+
+  // TODO (bvaughn) Replace this behavior with an invariant() in the future.
+  // It has only been added in Fiber to match the (unintentional) behavior in Stack.
+  if (typeof instance.getChildContext !== 'function') {
+    warning(
+      false,
```

Comment:
Here is an example of how we do it https://github.com/facebook/react/pull/8739.

This PR would be much simpler though. Since warning is specific to a class, we can use component name as the key in the `warnedAboutMissingChildContext` cache.